test/new_relic/agent/llm_test.rb

```diff
@@ -0,0 +1,132 @@
+          parent_segment = NewRelic::Agent::Transaction::AbstractSegment.new('Llm/embedding/OpenAI/embeddings')
+          segment = NewRelic::Agent::Transaction::AbstractSegment.new('Anything')
+          segment.parent = parent_segment
+
+          assert_truthy NewRelic::Agent::LLM.openai_parent?(segment)
+        end
+      end
+
+      def test_openai_parent_without_segment
+        NewRelic::Agent.stub(:config, {:'instrumentation.ruby_openai' => :auto, :'ai_monitoring.enabled' => true}) do
+          refute NewRelic::Agent::LLM.openai_parent?(nil)
+        end
+      end
+
+      def test_opneai_parent_without_parent
+        NewRelic::Agent.stub(:config, {:'instrumentation.ruby_openai' => :auto, :'ai_monitoring.enabled' => true}) do
+          segment = NewRelic::Agent::Transaction::AbstractSegment.new('Anything')
+
+          refute NewRelic::Agent::LLM.openai_parent?(segment)
+        end
+      end
+
+      def test_opneai_parent_without_name
+        NewRelic::Agent.stub(:config, {:'instrumentation.ruby_openai' => :auto, :'ai_monitoring.enabled' => true}) do
+          parent_segment = NewRelic::Agent::Transaction::AbstractSegment.new(nil)
+          segment = NewRelic::Agent::Transaction::AbstractSegment.new('Anything')
+          segment.parent = parent_segment
+
+          refute NewRelic::Agent::LLM.openai_parent?(segment)
+        end
+      end
+
+      def test_opneai_parent_without_match
+        NewRelic::Agent.stub(:config, {:'instrumentation.ruby_openai' => :auto, :'ai_monitoring.enabled' => true}) do
+          parent_segment = NewRelic::Agent::Transaction::AbstractSegment.new('doesnt_match')
+          segment = NewRelic::Agent::Transaction::AbstractSegment.new('Anything')
+          segment.parent = parent_segment
+
+          refute NewRelic::Agent::LLM.openai_parent?(segment)
+        end
+      end
+
+      def test_populate_openai_response_headers_without_llm_event
+        NewRelic::Agent.stub(:config, {:'instrumentation.ruby_openai' => :auto, :'ai_monitoring.enabled' => true}) do
+          response = {}
+          parent = NewRelic::Agent::Transaction::AbstractSegment.new('Llm/embedding/OpenAI/embeddings')
+
+          refute parent.instance_variable_defined?(:@llm_event)
+          assert_nil NewRelic::Agent::LLM.populate_openai_response_headers(response, parent)
+        end
+      end
+
+      def test_populate_openai_response_headers_with_llm_event_calls_llm_method
+        NewRelic::Agent.stub(:config, {:'instrumentation.ruby_openai' => :auto, :'ai_monitoring.enabled' => true}) do
+          response = {}
+          parent = NewRelic::Agent::Transaction::AbstractSegment.new('Llm/embedding/OpenAI/embeddings')
+          mock_llm_event = NewRelic::Agent::Llm::Embedding.new
+          mock_llm_event.expects(:populate_openai_response_headers).with(response)
+          parent.llm_event = mock_llm_event
+          NewRelic::Agent::LLM.populate_openai_response_headers(response, parent)
```

Comment:
I remember pairing on this one. It's not immediately clear to someone not readily familiar with Mocha, but the `#expects` call is essentially performing a verification on the mock. So that's ultimately where the value is coming from here. Is there really nothing else we can hang an assert on though? Does `response` get modified or anything else take place that we could confirm?

Reply:
We might be able to refactor this to check to see if headers get assigned to the llm event. 

How does this look? 

```ruby
      def test_populate_openai_response_headers_with_llm_event_adds_headers_to_llm_event
        NewRelic::Agent.stub(:config, {:'instrumentation.ruby_openai' => :auto, :'ai_monitoring.enabled' => true}) do
          request_id = 'PIR8SH1P'
          response = {'x-request-id' => [request_id]}
          parent = NewRelic::Agent::Transaction::AbstractSegment.new('Llm/embedding/OpenAI/embeddings')
          mock_llm_event = NewRelic::Agent::Llm::Embedding.new
          parent.llm_event = mock_llm_event
          NewRelic::Agent::LLM.populate_openai_response_headers(response, parent)

          assert_equal request_id, mock_llm_event.request_id
        end
      end
```

Reply:
> How does this look?

I personally love this, because:

- It's now easier to intuit what is taking place
- We now have an assertion, which I think is nicer than the `expects` we had previously
- We can now remove the `teardown` method and Mocha dependency from this test class entirely

Reply:
Wonderful! Applied in 7bf893aa65ef17d151db76d8e9aaf75957b34c78

Reply:
I like this^